lib/logstash/inputs/redis.rb

```diff
@@ -248,18 +240,20 @@ def list_single_listener(redis, output_queue)
 
   # private
   def subscribe_stop
-    return if @redis.nil? || !@redis.connected?
-    # if its a SubscribedClient then:
-    # it does not have a disconnect method (yet)
-    if @redis.subscribed?
+    redis = @redis # might change during method invocation
+    return if redis.nil? || !redis.connected?
+
+    if redis.subscribed?
       if @data_type == 'pattern_channel'
-        @redis.punsubscribe
+        redis.punsubscribe
       else
-        @redis.unsubscribe
+        redis.unsubscribe
       end
     else
-      @redis.disconnect!
+      redis.disconnect!
```

Comment:
This commit removes this comment:

```
 if its a SubscribedClient then:
    # it does not have a disconnect method (yet)
```

It feels like there is some context missing as to why `disconnect!` is not called when redis is subscribed, if this is indeed still the desired behaviour.

Reply:
believe this was a left over from the Redis 3.x days. 
[`Redis#disconnect`](https://github.com/redis/redis-rb/blob/v4.0.1/lib/redis.rb#L67) is always fine to call (in 4.x) regardless of the `@client` used (`@original_client` is never "replaced").

also no reason not to have disconnect (`close`) here, I've updated the PR - thanks for the hint.